content/docs/nextjs/github-public-repo.md

```diff
@@ -0,0 +1,309 @@
+BASE_BRANCH=master
+```
+
+To load these .env values in the front-end, your `next.config` will need to be configured. We will also use the `dotenv` package to load our `.env` file:
+
+`npm install --save dotenv`
+
+```
+// next.config.js
+require("dotenv").config();
+
+module.exports = {
+  env: {
+    GITHUB_CLIENT_ID: process.env.GITHUB_CLIENT_ID,
+    REPO_FULL_NAME: process.env.REPO_FULL_NAME,
+    BASE_BRANCH: process.env.BASE_BRANCH,
+  },
+  // ...
+}
+```
+
+For help setting up environment variables with Next, see the [Next docs](https://nextjs.org/docs/api-reference/next.config.js/environment-variables)
+
+## Setup
+
+### Register the GitHubClient
+
+We will want to use the GitHubClient to load/save our content using the GitHub API. Let's add it as an API plugin.
+
+```ts
+// pages/_app.js
+import { TinaCMS } from 'tinacms'
+import { GitHubClient } from 'react-tinacms-github'
+
+const REPO_FULL_NAME = process.env.REPO_FULL_NAME as string // e.g: tinacms/tinacms.org
+
+const cms = new TinaCMS({
+  apis: {
+    github: new GitHubClient('/api/proxy-github', REPO_FULL_NAME),
+  },
+  // ... any other tina config
+})
+```
+
+You'll notice that the GitHubClient takes in a string ('/api/proxy-github' in our case). All requests using the GitHubClient gets passed through a custom proxy, so that we can attach the authentication tokens on the backend.
+
+Let's setup up some of these backend API functions.
+
+## API functions
+
+We will need a few API functions to handle GitHub authentication, and Next.js's preview-mode:
+With Next.js, any functions in the `pages/api` directory are are mapped to `/api/*` endpoints.
+
+We've created a script to make generating these files easier:
+
+From the console, run:
+
+```
+$ npx hygen-add https://github.com/dwalkr/hygen-next-tinacms-github
+$ npx hygen next-tinacms-github bootstrap --format ts
+```
+
+_Note: if your **/pages** directory in not in the root, you will need to supply a `--dir [subDir]` option._
+
+```
+$ npx hygen next-tinacms-github bootstrap --format ts --dir src
+```
+
+You'll see that a few API functions have been setup for us.
+
+#### `preview.ts`
+
+Contains API function to enter preview-mode, and set the preview-data with content stored in the cookies.
+
+#### `proxy-github.ts`
+
+Contains API function to attach the user's auth token, and proxy requests to the GitHub API.
+
+#### `create-github-auth-token.ts`
+
+Helper for creating a `createCreateAccessToken` server function.
+
+> _Note: You may need to configure the `process.env.GITHUB_CLIENT_ID` and `process.env.GITHUB_CLIENT_SECRET` environment variables within this file._ _See [Next's documentation](https://nextjs.org/docs/api-reference/next.config.js/environment-variables) for adding environment variables_
+
+> [See below](#github-oauth-app) for instructions on creating a GitHub OAuth App to generate these **Client ID** & **Client Secret** variables.
```

Comment:
We already setup the .env file in an earlier step, so I don't think this is needed
```suggestion
```